### PR TITLE
Improve entity position updates

### PR DESCRIPTION
**Pull request by tigerw**
* Move determination of whether a delta is too big for a packet into the protocol handlers
+ Less jittery movement
+ Generalise CollectEntity to take any entity

### CODE REVIEW

**Comment by tigerw** [comment]
As a result pickups move somewhat nicer than before.

**Comment by tigerw** [comment]
Before
Before
After
After

File: src/Entities/Boat.cpp [hunk 1]
```diff
@@ -49,19 +49,11 @@ void cBoat::BroadcastMovementUpdate(const cClientHandle * a_Exclude)
...
-		if ((abs(Diff.x) <= 127) && (abs(Diff.y) <= 127) && (abs(Diff.z) <= 127))  // Limitations of a Byte
-		{
-			m_World->BroadcastEntityRelMove(*this, Vector3<Int8>(Diff), a_Exclude);
-		}
-		else
-		{
-			// Too big a movement, do a teleport
-			m_World->BroadcastTeleportEntity(*this, a_Exclude);
-		}
+		m_World->BroadcastEntityPosition(*this, a_Exclude);
```
[the remark this answers is not in the document]
Comment (mathiascode) [reply src/Entities/Boat.cpp:54]: https://github.com/cuberite/cuberite/pull/4488

[the remark this answers is not in the document]
Comment (bearbin) [reply src/Entities/Boat.cpp:54]: It still needs a comment though.

**Pull request merged**